azkaban-common/src/main/java/azkaban/event/Event.java

```diff
@@ -53,24 +53,13 @@ public long getTime() {
     return time;
   }
 
-  public Object getData() {
+  public EventData getData() {
     return eventData;
   }
 
-  public static Event create(Object runner, Type type) {
-    return new Event(runner, type, null, true);
-  }
-
-  public static Event create(Object runner, Type type, Object eventData) {
-    return new Event(runner, type, eventData, true);
+  public static Event create(Object runner, Type type, EventData eventData) {
+    Preconditions.checkNotNull(eventData, "EventData was null");
```

Comment:
I just modified this method signature. Code wouldn't compile if I hand't modified _all_ its callers. So I did, and made sure that it isn't called with null. If someone adds a new call to `Event.create`, they can test that it works?

I can remove the null-check here if you want, but it will be then much harder to debug where the null came from if NPE happens later in an asynchronous callback. At that point there won't be a useful stacktrace to point who created the event with a null value. What do you prefer?